heat/core/manipulations.py

```diff
@@ -1735,6 +1796,42 @@ def reshape_argsort_counts_displs(
         displs[1:] = torch.cumsum(counts[:-1], dim=0)
         return argsort, counts, displs
 
+    if shape == -1:
+        shape = (a.gnumel,)
+
+    if not isinstance(shape, (list, tuple)):
+        raise TypeError("shape must be list, tuple, currently {}".format(type(shape)))
+
+    # check new_split parameter
+    if new_split is None:
+        new_split = a.split
+    stride_tricks.sanitize_axis(shape, new_split)
+
+    # Check the type of shape and number elements
+    shape = list(shape)
+    for dim in shape:
+        if not isinstance(dim, int):
+            raise TypeError(
+                "argument 'shape' contains an element of type {}, only ints are allowed".format(
+                    type(dim)
+                )
+            )
+        if dim < -1:
+            raise ValueError("negative dimensions are not allowed")
+
+    # infer unknown dimension
+    if shape.count(-1) > 1:
+        raise ValueError("too many unknown dimensions")
+    elif shape.count(-1) == 1:
+        pos = shape.index(-1)
+        shape[pos] = -(
+            a.size / torch.prod(torch.tensor(shape, dtype=torch.int, device=tdevice))
+        ).item()
```

Comment:
Again, just a suggestions, but could we have the checks in order of decreasing likelihood of condition? As in:
```
if shape.count(-1) == 0:
   check prod(shape) vs. a.size
elif shape.count(-1) == 1:
   infer unknown dimension
else:
   complain: too many unknown dimensions 

```